@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,190 @@ export namespace MapViewUtils {
+        // generally small, the difference in normals should be for most cases very small and the
+        // extra computation required to compute such a small value isn't worth it. The value we
+        // compute is anyway conservative, so we don't risk computing something that will put the
+        // camera outside the valid range.
+        const maxTilt = ninetyRad + angleBetweenNormals - maxRotationTargetAngle;
+
+        const clampedDeltaTiltMapTarget =
+            MathUtils.clamp(deltaTilt + currentTiltRotationCenter, 0, maxTilt) -
+            currentTiltRotationCenter;
+
+        const maxRotationTiltAngle = THREE.MathUtils.degToRad(89);
+        // Here we clamp to 89 degrees, to prevent it intersecting with the world at 90, note, we
+        // don't use the `maxTiltAngle`, because then special handling needs to be applied when for
+        // example the tilt at the rotation center is more than that at the map target, then the
+        // tilt feels different at the center / offset positions. Using 89 gives the most natual
+        // feeling when applying the tilt angle.
+        const clampedDeltaTiltRotationTarget =
+            MathUtils.clamp(deltaTilt + currentTiltRotationCenter, 0, maxRotationTiltAngle) -
+            currentTiltRotationCenter;
+
+        // Math.min doesn't work, because we need to keep the sign of the delta as well.
         const clampedDeltaTilt =
-            MathUtils.clamp(deltaTilt + currentTilt, 0, maxTiltAngle) - currentTilt;
-        if (Math.abs(clampedDeltaTilt) <= Number.EPSILON) {
-            return;
+            Math.abs(clampedDeltaTiltMapTarget) < Math.abs(clampedDeltaTiltRotationTarget)
+                ? clampedDeltaTiltMapTarget
+                : clampedDeltaTiltRotationTarget;
+
+        // It can happen in some cases that the maxRotationTiltAngle is exceeded because the
+        // `clampedDeltaTilt` is taken from the smaller value clampedDeltaTiltMapTarget. We check
+        // that we don't exceed 0, because otherwise it will snap to 89, so we let the user ease
+        // back to 89, but not any further.
+        if (
+            currentTiltRotationCenter + clampedDeltaTilt >= maxRotationTiltAngle &&
+            clampedDeltaTilt > 0
+        ) {
+            return maxRotationTiltAngle - currentTiltRotationCenter;
         }
 
-        const posBackup = cache.vector3[1].copy(camera.position);
+        return clampedDeltaTilt;
+    }
+
+    /**
+     * @hidden
+     * @internal
+     *
+     * Applies the given tilt to the camera around the supplied target.
+     */
+    function applyTiltAroundTarget(
+        mapView: MapView,
+        rotationTargetWorld: THREE.Vector3,
+        deltaTilt: number
+    ) {
+        const camera = mapView.camera;
+        // We use index 2, because these values are required if the method: rayCastWorldCoordinates
+        // fails, which itself uses indexes 0 and 1.
+        const posBackup = cache.vector3[2].copy(camera.position);
```

Comment:
These problems with functions in Utils.ts trashing each other's temp variables are getting scary.
We are abusing temp caches in this file, having a cache object of temp variable arrays is just too much.

If you had to add a comment to clarify why you have to use that tmp variable and no other, it's a good sign we have to stop. I suggest you just clone camera position. If later we see a problem with that, we'll see what we can do.

I would vote to dismantle that cache object we use here.

Reply:
Fully, agree, it is a nightmare to program with, I had numerous times when bugs showed just because I forgot I was overwriting memory with a function call that I needed later.

Will remove them in this case that you mention. In the other cases, the cache usage is ok, because something is calculated immediately and not required anymore.